.github/workflows/release-go-functions.yaml

```diff
@@ -0,0 +1,31 @@
+name: release-go-functions
+
+on:
+  push:
+    tags:
+      - release-go-functions-*
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
I added a Makefile, and the call of `make` into `functions/go/publish-functions.sh` (instead of github workflow), since every go function directory will need to run `make`.